base/repl/LineEdit.jl

```diff
@@ -1592,6 +1618,23 @@ buffer(s::PromptState) = s.input_buffer
 buffer(s::SearchState) = s.query_buffer
 buffer(s::PrefixSearchState) = s.response_buffer
 
+function empty_undo(s::PromptState)
+    empty!(s.undo_buffers)
+end
+empty_undo(s) = nothing
+
+function push_undo(s::PromptState)
```

Comment:
These should probably all be refactored to have `!`. It wasn't clear to me reading through this that it actually modified the `PromptState` until I saw the definition. The larger refactoring doesn't have to happen in this PR, but if you're adding a function now, might as well use the standard notation, IMO.